pyanaconda/storage/root.py

```diff
@@ -312,32 +312,42 @@ def _parse_fstab(devicetree, chroot):
 class Root(object):
     """A root represents an existing OS installation."""
 
-    def __init__(self, mounts=None, swaps=None, name=None):
-        """
-            :keyword mounts: mountpoint dict
-            :type mounts: dict (mountpoint keys and :class:`blivet.devices.StorageDevice` values)
-            :keyword swaps: swap device list
-            :type swaps: list of :class:`blivet.devices.StorageDevice`
-            :keyword name: name for this installed OS
-            :type name: str
+    def __init__(self, name=None, product=None, version=None, arch=None, mounts=None, swaps=None):
+        """Create a new OS representation.
+
+        :param name: a name of the OS or None
+        :param product: a distribution name or None
+        :param version: a distribution version or None
+        :param arch: a machine's architecture or None
+        :param mounts: a dictionary of mount points and devices
+        :param swaps: a list of swap devices
         """
-        # mountpoint key, StorageDevice value
-        if not mounts:
-            self.mounts = {}
-        else:
-            self.mounts = mounts
+        self._name = name
+        self._product = product
+        self._version = version
+        self._arch = arch
 
-        # StorageDevice
-        if not swaps:
-            self.swaps = []
-        else:
-            self.swaps = swaps
+        # Blivet needs to be able to set these attributes.
+        self.mounts = mounts or {}
+        self.swaps = swaps or []
 
-        self.name = name    # eg: "Fedora Linux 16 for x86_64", "Linux on sda2"
+    @property
+    def name(self):
+        """The name of the OS."""
+        if self._name:
+            return self._name
 
-        if not self.name and "/" in self.mounts:
-            self.name = self.mounts["/"].format.uuid
```

Comment:
I'm missing this part in the new code. Did I overlooked or is it removed on purpose?

Reply:
The name was always specified to something, so this code was never used.